x-pack/plugins/licensing/server/plugin.ts

```diff
@@ -180,16 +175,17 @@ export class LicensingPlugin implements Plugin<LicensingPluginSetup, LicensingPl
     };
   }
 
-  private fetchLicense = async (clusterClient: ILegacyClusterClient): Promise<ILicense> => {
+  private fetchLicense = async (clusterClient: MaybePromise<IClusterClient>): Promise<ILicense> => {
+    const client = isPromise(clusterClient) ? await clusterClient : clusterClient;
     try {
-      const response = await clusterClient.callAsInternalUser('transport.request', {
-        method: 'GET',
-        path: '/_xpack?accept_enterprise=true',
+      const { body: response } = await client.asInternalUser.xpack.info({
+        // @ts-expect-error `accept_enterprise` is not present in the client definition
+        accept_enterprise: true,
       });
-
-      const normalizedLicense = response.license
-        ? normalizeServerLicense(response.license)
-        : undefined;
+      const normalizedLicense =
+        response.license && response.license.type !== 'missing'
```

Comment:
TIL there is a `missing` type in the client def. Not sure this is really used, as it seems we never encountered it, but still added the check.